packages/teams-js/src/public/meeting.ts

```diff
@@ -560,4 +601,121 @@ export namespace meeting {
+    callback: (isHostAudioless: boolean) => void,
+  ): void {
+    const callbackInternalRequest = (error: SdkError | null, isHostAudioless: boolean | null): void => {
+      if (error && isHostAudioless != null) {
+        throw new Error('[requestAppAudioHandling] Callback response - both parameters cannot be set');
+      }
+      if (error) {
+        throw new Error(`[requestAppAudioHandling] Callback response - SDK error ${error.errorCode} ${error.message}`);
+      }
+      if (typeof isHostAudioless !== 'boolean') {
+        throw new Error('[requestAppAudioHandling] Callback response - isHostAudioless must be a boolean');
+      }
+
+      const micStateChangedCallback = async (micState: MicState): Promise<void> => {
+        const newMicState = await requestAppAudioHandlingParams.callbackMicMuteStateChangedHandler(micState);
+
+        const micStateDidUpdate = newMicState.isMicMuted === micState.isMicMuted;
+        setMicStateWithReason(
+          newMicState,
+          micStateDidUpdate ? MicStateChangeReason.HostInitiated : MicStateChangeReason.AppDeclinedToChange,
+        );
+      };
+      registerHandler('meeting.micStateChanged', micStateChangedCallback);
+
+      callback(isHostAudioless);
+    };
+    sendMessageToParent(
+      'meeting.requestAppAudioHandling',
+      [requestAppAudioHandlingParams.isAppHandlingAudio],
+      callbackInternalRequest,
+    );
+  }
+
+  function stopAppAudioHandling(
+    requestAppAudioHandlingParams: RequestAppAudioHandlingParams,
+    callback: (isHostAudioless: boolean) => void,
+  ): void {
+    const callbackInternalStop = (error: SdkError | null, isHostAudioless: boolean | null): void => {
+      if (error && isHostAudioless != null) {
+        throw new Error('[requestAppAudioHandling] Callback response - both parameters cannot be set');
+      }
+      if (error) {
+        throw new Error(`[requestAppAudioHandling] Callback response - SDK error ${error.errorCode} ${error.message}`);
+      }
+      if (typeof isHostAudioless !== 'boolean') {
+        throw new Error('[requestAppAudioHandling] Callback response - isHostAudioless must be a boolean');
+      }
+
+      if (doesHandlerExist('meeting.micStateChanged')) {
+        removeHandler('meeting.micStateChanged');
+      }
+
+      callback(isHostAudioless);
+    };
+
+    sendMessageToParent(
+      'meeting.requestAppAudioHandling',
+      [requestAppAudioHandlingParams.isAppHandlingAudio],
+      callbackInternalStop,
+    );
+  }
+
+  /**
+   * Notifies the host that the microphone state has changed in the app.
+   * @param micState - The new state that the microphone is in
+   *   isMicMuted - Boolean to indicate the current mute status of the mic.
+   *
+   * @beta
+   */
+  export function updateMicState(micState: MicState): void {
+    setMicStateWithReason(micState, MicStateChangeReason.AppInitiated);
+  }
+
+  function setMicStateWithReason(micState: MicState, reason: MicStateChangeReason): void {
```

Comment:
func name and message name being sent are not aligned.
Better to have aligned name?

Reply:
Makes sense, how does `sendMicMuteStatusResponseWithReason` for the function name sound? Or should we change the message name?

Reply:
Or how about changing the message name to `meeting.updateMicState`, since it's not necessarily sent as a response

Reply:
I changed it, lmk if you think of another name you'd rather use